src/main.cpp

```diff
@@ -203,393 +150,226 @@ void run()
-            ((mibibytes) ? std::pow(2.0, -20.0) : 1.0E-6) * sizes[i] / (*minmax.first)
-          << std::left << std::setw(12) << std::setprecision(5) << *minmax.first
-          << std::left << std::setw(12) << std::setprecision(5) << *minmax.second
-          << std::left << std::setw(12) << std::setprecision(5) << average
-          << std::endl;
-      }
-    }
-  } else if (selection == Benchmark::Triad)
-  {
-    // Display timing results
-    double total_bytes = 3 * sizeof(T) * ARRAY_SIZE * num_times;
-    double bandwidth = ((mibibytes) ? std::pow(2.0, -30.0) : 1.0E-9) * (total_bytes / timings[0][0]);
+    // Calculate average; ignore the first result
+    double average = std::accumulate(timings[i].begin()+1, timings[i].end(), 0.0)
+      / (double)(num_times - 1);
 
-    if (output_as_csv)
-    {
-      std::cout
-        << "function" << csv_separator
-        << "num_times" << csv_separator
-        << "n_elements" << csv_separator
-        << "sizeof" << csv_separator
-        << ((mibibytes) ? "gibytes_per_sec" : "gbytes_per_sec") << csv_separator
-        << "runtime"
-        << std::endl;
-      std::cout
-        << "Triad" << csv_separator
-        << num_times << csv_separator
-        << ARRAY_SIZE << csv_separator
-        << sizeof(T) << csv_separator
-        << bandwidth << csv_separator
-        << timings[0][0]
-        << std::endl;
-    }
-    else
-    {
-      std::cout
-        << "--------------------------------"
-        << std::endl << std::fixed
-        << "Runtime (seconds): " << std::left << std::setprecision(5)
-        << timings[0][0] << std::endl
-        << "Bandwidth (" << ((mibibytes) ? "GiB/s" : "GB/s") << "):  "
-        << std::left << std::setprecision(3)
-        << bandwidth << std::endl;
-    }
+    // Display results
+    fmt_result(bench[i].label, num_times, ARRAY_SIZE, sizeof(T),
+	       fmt_bw(bench[i].weight, *minmax.first), *minmax.first, *minmax.second, average);
   }
-
-  delete stream;
-
 }
 
-
 template <typename T>
-void check_solution(const unsigned int ntimes, std::vector<T>& a, std::vector<T>& b, std::vector<T>& c, T& sum)
+void check_solution(const size_t num_times,
+		    std::vector<T>& a, std::vector<T>& b, std::vector<T>& c, T& sum)
 {
   // Generate correct solution
   T goldA = startA;
   T goldB = startB;
   T goldC = startC;
-  T goldSum{};
+  T goldS = T(0.);
 
   const T scalar = startScalar;
 
-  for (unsigned int i = 0; i < ntimes; i++)
+  for (size_t b = 0; b < num_benchmarks; ++b)
   {
-    // Do STREAM!
-    if (selection == Benchmark::All)
-    {
-      goldC = goldA;
-      goldB = scalar * goldC;
-      goldC = goldA + goldB;
-      goldA = goldB + scalar * goldC;
-    } else if (selection == Benchmark::Triad)
-    {
-      goldA = goldB + scalar * goldC;
-    } else if (selection == Benchmark::Nstream)
+    if (!run_benchmark(bench[b])) continue;
+
+    for (size_t i = 0; i < num_times; i++)
```

Comment:
As above, I think the loop order needs switching for `check_solution()` does the arithmetic in the same order.